server/src/main/java/org/elasticsearch/search/query/QueryPhaseCollector.java

```diff
@@ -52,10 +49,6 @@ public final class QueryPhaseCollector implements Collector {
     private final boolean cacheScores;
     private boolean terminatedAfter = false;
 
-    QueryPhaseCollector(Collector topDocsCollector, Weight postFilterWeight, int terminateAfter, Collector aggsCollector, Float minScore) {
-        this(topDocsCollector, postFilterWeight, resolveTerminateAfterChecker(terminateAfter), aggsCollector, minScore);
-    }
```

Comment:
Removed this as it was needed only for sequential scenarios. In fact the terminate after checker needs to be shared across the different collectors that the same instance of the collector manager creates.